### PR TITLE
Add try/except for expected null-schema error in read_parquet

### PR DESCRIPTION
**Pull request by rjzamora**
## Description
Temporary workaround for https://github.com/rapidsai/cudf/issues/12702 - Adds a try/except to concatenate distinct `cudf.read_parquet` results instead of requiring file aggregation to occur within libcudf.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by wence- — CHANGES_REQUESTED** [review]
Can we test that the test is successful please? Otherwise looks good.

File: python/dask_cudf/dask_cudf/io/tests/test_parquet.py [hunk 1]
```diff
@@ -502,3 +502,15 @@ def test_check_file_size(tmpdir):
...
+def test_nullable_schema_mismatch(tmpdir):
+    # See: https://github.com/rapidsai/cudf/issues/12702
+    path0 = str(tmpdir.join("test.0.parquet"))
+    path1 = str(tmpdir.join("test.1.parquet"))
+    cudf.DataFrame.from_dict({"a": [1, 2, 3]}).to_parquet(path0)
+    cudf.DataFrame.from_dict({"a": [4, 5, None]}).to_parquet(path1)
+    with dask.config.set({"dataframe.backend": "cudf"}):
+        dd.read_parquet(
+            [path0, path1], split_row_groups=2, aggregate_files=True
+        ).compute()
```
[the remark this answers is not in the document]
Comment (rjzamora) [reply python/dask_cudf/dask_cudf/io/tests/test_parquet.py]: yup - makes sense

**Review by wence- — APPROVED** [review]
Thanks!

**Pull request merged**